mrchecker-docs/documentation/Who-Is-MrChecker/Test-Framework-Modules/Core-Test-Module-Different-Environments.asciidoc

```diff
@@ -1,150 +1,133 @@
+:tip-caption: :bulb:
+:note-caption: :information_source:
+:important-caption: :heavy_exclamation_mark:
+:caution-caption: :fire:
+:warning-caption: :warning:
+endif::[]
+
+toc::[]
+:idprefix:
+:idseparator: -
+:reproducible:
+:source-highlighter: rouge
+:listing-caption: Listing
+
+
+= System under test environments
+
+image::images/image080.png[]
+
+* *Quality assurance* or *QA* is a way of preventing mistakes or defects in the manufactured products and avoiding problems when delivering solutions or services to customers which ISO 9000 defines as "part of quality management focused on providing confidence that quality requirements will be fulfilled".
+
+* *System integration testing* or *SIT* is a high-level software testing process in which testers verify that all related systems maintain data integrity and can operate in coordination with other systems in the same environment. The testing process ensures that all sub-components are integrated successfully to provide expected results.
+
+* *Development* or *Dev* testing is performed by the software developer or engineer during the construction phase of the software development life-cycle. Rather than replace traditional QA focuses, it augments it. Development testing aims to eliminate construction errors before code is promoted to QA; this strategy is intended to increase the quality of the resulting software as well as the efficiency of the overall development and QA process. 
+
+* *Prod* if the customer accepts the product, it is deployed to a *production* environment, making it available to all users of the system.
+
+image::images/image051.png[]
+
+= How to use system environment
+
+In Page classes, when you load / start web, it is uncommon to save fixed main url.  
+
+Value flexibility is a must, when your web application under test has different main url, depending on the environmnent (DEV, QA, SIT, ..., PROD) 
+
+Instead of *hard coded* main url variable, you build your Page classes with dynamic variable. 
+
+An example of dynamic variable *GetEnvironmentParam.WWW_FONT_URL*
+
+image::images/image081.png[]
+
+= How to create / update system environment
+
+== External file with variable values 
+
+Dynamic variable values are stored under  *mrchecker-app-under-test\src\resources\enviroments\environments.csv*. 
+
+**NOTE:** As environments.csv is a comma-separated file, please be careful while editing and then save it under Excel. 
+
+image::images/image082.png[]
+
+== Encrypting sensitive data
+
+Some types of data you might want to store as environment settings are sensitive in nature (e.g. passwords). You might not want to store them (at least not in their plaintext form) in your repository. To be able to encrypt sensitive data you need to do following:
+
+* 1 Create a secret (long, random chain of characters) and store it under *mrchecker-app-under-test\src\resources\secretData.txt*. Example: `LhwbTm9V3FUbBO5Tt5PiTUEQrXGgWrDLCMthnzLKNy1zA5FVTFiTdHRQAyPRIGXmsAjPUPlJSoSLeSBM`

```

Comment:
Decide on one type of bullets * or 1.
Path to file should be mrchecker-app-under-test\src\resources\secretData

Reply:
I replaced all long paths/links to git with "link:", now i'll try to change it back to 1., there was a problem with this last time, it was continuing from 4. instead of 1.

Reply:
Changed list to *